server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java

```diff
@@ -202,12 +184,14 @@ public CommonStats getPrimaries() {
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeArray(shards);
-        if (out.getTransportVersion().onOrAfter(TransportVersions.INDEX_STATS_ADDITIONAL_FIELDS)) {
+        if (out.getTransportVersion().onOrAfter(TransportVersions.INDEX_STATS_ADDITIONAL_FIELDS_REVERT)) {
             out.writeMap(indexHealthMap, StreamOutput::writeWriteable);
             out.writeMap(indexStateMap, StreamOutput::writeWriteable);
-            out.writeMap(indexTierPreferenceMap, StreamOutput::writeStringCollection);
-            out.writeMap(indexCreationDateMap, StreamOutput::writeLong);
-
+        } else if (out.getTransportVersion().onOrAfter(TransportVersions.INDEX_STATS_ADDITIONAL_FIELDS)) {
+            out.writeMap(indexHealthMap, StreamOutput::writeWriteable);
+            out.writeMap(indexStateMap, StreamOutput::writeWriteable);
+            out.writeMap(Map.of(), StreamOutput::writeStringCollection);
+            out.writeMap(Map.of(), StreamOutput::writeLong);
```

Comment:
Should we add some kind of check to be able to _eventually_ remove these?

Reply:
Yeah eventually all these transport versions get collapsed into a release-version-named one (see e.g. `org.elasticsearch.TransportVersions#V_8_15_2`) and then this branch will be dead code:

```
        if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_16_x)) {
            indexHealthMap = in.readMap(ClusterHealthStatus::readFrom);
            indexStateMap = in.readMap(IndexMetadata.State::readFrom);
        } else if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_16_x)) {
            // same condition as the previous if so cannot get here
```